### PR TITLE
add from_buffer method to Frame

### PR DESCRIPTION
**Pull request by rukai**
This sync implementation is also useful for async applications that make use of https://docs.rs/tokio-util/0.6.9/tokio_util/codec/struct.FramedRead.html
In particular for implementing https://docs.rs/tokio-util/0.6.9/tokio_util/codec/trait.Decoder.html#tymethod.decode which is not async.

Unfortunately I couldnt see a good way to share code with the existing async implementation.
Possibly this should supercede the async implementation one day but for now I think they can just coexist.

### CODE REVIEW

File: cassandra-protocol/src/frame.rs [hunk 1]
```diff
@@ -117,6 +123,85 @@ impl Frame {
...
+    /// * `Err(_)` - The frame is malformed and you should close the connection as this method does not provide a way to tell how many bytes to advance the buffer in this case.
+    pub fn from_buffer(
+        data: &[u8],
+        compressor: Compression,
+    ) -> Result<(Frame, usize), ParseFrameError> {
+        if data.len() < HEADER_LEN {
+            return Err(ParseFrameError::NotEnoughBytes);
+        }
+
+        let body_len = try_i32_from_bytes(&data[5..9]).unwrap() as usize;
```
Comment (krojew) [inline cassandra-protocol/src/frame.rs:142]: Maybe return an error here instead of panicking.

File: cassandra-protocol/src/frame.rs [hunk 2]
```diff
@@ -117,6 +123,85 @@ impl Frame {
...
+        let frame_len = HEADER_LEN + body_len;
+        if data.len() < frame_len {
+            return Err(ParseFrameError::NotEnoughBytes);
+        }
+
+        let version = Version::try_from(data[0])
+            .map_err(|_| ParseFrameError::UnsupportedVersion(data[0] & 0x7f))?;
+        let direction = Direction::from(data[0]);
+        let flags = Flags::from_bits_truncate(data[1]);
+        let stream = try_i16_from_bytes(&data[2..4]).unwrap();
```
Comment (krojew) [inline cassandra-protocol/src/frame.rs:152]: As above.

**Review by krojew — CHANGES_REQUESTED** [review]

File: cassandra-protocol/src/frame.rs (hunk 1 above)
Comment (rukai) [reply cassandra-protocol/src/frame.rs:142]: An error there is impossible.
a valid i32 can always be constructed from 4 bytes.

Comment (krojew) [reply cassandra-protocol/src/frame.rs:142]: True. It would be great to avoid going to and from a slice along the way.

**Review by krojew — APPROVED** [review]

**Pull request merged**